runtime/src/bank/builtins/core_bpf_migration/mod.rs

```diff
@@ -364,6 +371,91 @@ impl Bank {
         Ok(())
     }
 
+    /// Migrate a Loader v2 BPF program.
+    ///
+    /// To use this function, add a feature-gated callsite to bank's
+    /// `apply_feature_activations` function, similar to below.
+    ///
+    /// ```ignore
+    /// if new_feature_activations.contains(&agave_feature_set::test_upgrade_program::id()) {
+    ///     self.migrate_bpf_loader_v2_to_v3(
+    ///        &bpf_loader_v2_program_address,
+    ///        &source_buffer_address,
+    ///        "test_upgrade_loader_v2_bpf_program",
+    ///     );
+    /// }
+    /// ```
+    /// The `source_buffer_address` must point to a Loader v3 buffer account
+    /// (state equal to [`UpgradeableLoaderState::Buffer`]).
+    #[allow(dead_code)] // Only used when an upgrade is configured.
+    pub(crate) fn migrate_bpf_loader_v2_to_v3(
```

Comment:
I'm a little stuck on this function name. Including the term "BPF Loader" makes me think you're migrating the loader itself, and with it removed (ie. "Migrate v2 program to v3") it sounds like it's just a migration from one loader to another. However, we're changing the ELF completely in this process.

What do you think about these?

```suggestion
    pub(crate) fn replace_v2_program_with_v3_program(
```

```suggestion
    pub(crate) fn upgrade_v2_program_with_v3_program(
```

I'm open to suggestions, just thought I'd kick off some bikeshedding!

Reply:
Agree that the current name is not very clear.

Do you think we need to sneak in "bpf loader" somewhere? Without it might look like it is a "v2" version of the program, but it is migrating a program from one loader to another.

Reply:
To me I don't think it's a big deal, so whatever you think is best for that bit.

Reply:
What about `upgrade_loader_v2_program_with_loader_v3_program`? it is a bit long, but all the information is there. 😅 I am inclined to go with either:
* `upgrade_v2_program_with_v3_program`
* `upgrade_loader_v2_program_with_loader_v3_program`

Reply:
Haha yep, the super-long one wfm!